util/exponential-backoff.js

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const DEFAULT_MAX_RETRIES = 5
+const DEFAULY_RETRY_MULTIPLIER_MS = 100
+const DEFAULT_RETRY_COUNT = 0
+
+const logger = require('@adobe/aio-lib-core-logging')('@adobe/aio-lib-core-tvm', { provider: 'debug' })
+const fetch = require('node-fetch').default
+const util = require('util')
+const sleep = util.promisify(setTimeout)
+
+/**
+ * This function will retry connecting to a url end-point, with
+exponential backoff. Returns a Promise.
+ *
+ * @param {string} url endpoint url
+ * @param {object} requestOptions request options
+ * @param {object} retryOptions retry options with keys being maxRetries, retryMultipler and retryCount
+ * @returns {Promise} Promise object representing the http response
+ */
+async function exponentialBackoff (url, requestOptions, retryOptions) {
+  const {
+    maxRetries = DEFAULT_MAX_RETRIES,
+    retryMultiplier = DEFAULY_RETRY_MULTIPLIER_MS,
+    retryCount = DEFAULT_RETRY_COUNT
```

Comment:
@Himavanth, I think retryCount should not be exposed to the uset and should be an internal parameter to _exponentialBackoffHelper.
I also think that we should expose and use retryInterval time, any reason why you removed it?

Reply:
yes i just realized about retryCount and fixed in a different PR. retryInterval is still there. It's max retry interval that is removed which is essentially a duplicate of max retries.

Reply:
sry if I am missing something, by using the exponential-backoff function can I set the timeout interval?

Reply:
ya the multiplier is the interval which increases exponentially.

Reply:
Ok I guess we could add an interval param later if needed.. 
So for now I think we can have maxRetries & interval, I think retryCount should be internal and not exposed in the function interface